### DISCUSSION TITLE
Implementation of a spatially varying dde.Variable

### DISCUSSION THREAD

**Discussion by MaxGRudolph** [lululxvi/deepxde#761] [Q&A]
Hi all,

I wand to solve an inverse problem similar to the [1D diffusion example](https://deepxde.readthedocs.io/en/latest/demos/pinn_inverse/diffusion.1d.inverse.html) from the documentation.
However, I want to solve an inverse problem in 2D, having a spatially varying diffusion parameter, i.e., `dde.Variable(tensor[N_x, N_y])` or `dde.Variable(array[N_x, N_y])` in the 2D case, where `N_x, N_y` are the nummber of points for the parameter.

So how would I implement such a variable and make sure that the PDE is solved, taking the spatial variability into account?

Thanks!

**Comment by lululxvi** [comment]
See
- https://deepxde.readthedocs.io/en/latest/demos/pinn_inverse/elliptic.inverse.field.html
- https://github.com/lululxvi/deepxde/blob/master/examples/pinn_inverse/diffusion_reaction_rate.py
